torchaudio/datasets/yesno.py

```diff
@@ -17,10 +19,108 @@
+            "0_0_1_0_1_0_1_1.wav",
+            "0_0_1_1_0_0_0_1.wav",
+            "0_0_1_1_0_1_0_0.wav",
+            "0_0_1_1_0_1_1_0.wav",
+            "0_0_1_1_0_1_1_1.wav",
+            "0_0_1_1_1_0_0_0.wav",
+            "0_0_1_1_1_0_0_1.wav",
+            "0_0_1_1_1_1_0_0.wav",
+            "0_0_1_1_1_1_1_0.wav",
+            "0_1_0_0_0_1_0_0.wav",
+            "0_1_0_0_0_1_1_0.wav",
+            "0_1_0_0_1_0_1_0.wav",
+            "0_1_0_0_1_0_1_1.wav",
+            "0_1_0_1_0_0_0_0.wav",
+            "0_1_0_1_1_0_1_0.wav",
+            "0_1_0_1_1_1_0_0.wav",
+            "0_1_1_0_0_1_1_0.wav",
+            "0_1_1_0_0_1_1_1.wav",
+            "0_1_1_1_0_0_0_0.wav",
+            "0_1_1_1_0_0_1_0.wav",
+            "0_1_1_1_0_1_0_1.wav",
+            "0_1_1_1_1_0_1_0.wav",
+            "0_1_1_1_1_1_1_1.wav",
+            "1_0_0_0_0_0_0_0.wav",
+            "1_0_0_0_0_0_0_1.wav",
+            "1_0_0_0_0_0_1_1.wav",
+            "1_0_0_0_1_0_0_1.wav",
+            "1_0_0_1_0_1_1_1.wav",
+            "1_0_1_0_1_0_0_1.wav",
+            "1_0_1_1_0_1_1_1.wav",
+            "1_0_1_1_1_0_1_0.wav",
+            "1_0_1_1_1_1_0_1.wav",
+            "1_1_0_0_0_0_0_1.wav",
+            "1_1_0_0_0_1_1_1.wav",
+            "1_1_0_0_1_0_1_0.wav",
+            "1_1_0_0_1_0_1_1.wav",
+            "1_1_0_0_1_1_1_0.wav",
+            "1_1_0_1_0_1_0_0.wav",
+            "1_1_0_1_0_1_1_0.wav",
+            "1_1_0_1_1_0_0_1.wav",
+            "1_1_0_1_1_0_1_1.wav",
+            "1_1_0_1_1_1_1_0.wav",
+            "1_1_1_0_0_0_0_1.wav",
+            "1_1_1_0_0_1_0_1.wav",
+            "1_1_1_0_0_1_1_1.wav",
+            "1_1_1_0_1_0_1_0.wav",
+            "1_1_1_0_1_0_1_1.wav",
+            "1_1_1_1_0_0_1_0.wav",
+            "1_1_1_1_0_1_0_0.wav",
+            "1_1_1_1_1_0_0_0.wav",
+            "1_1_1_1_1_1_0_0.wav",
+            "1_1_1_1_1_1_1_1.wav",
+        ]
     }
 }
 
 
+@dataclass
+class YesNoItem:
+    path: str
+    label: List[int]
+    waveform: Tensor
+    sample_rate: int
+
+
+class ListYesNoItems(IterDataPipe):
+    """Given a root directory, return the list of files"""
+    def __init__(self, root):
+        self.data_dir = os.path.join(root, 'waves_yesno')
+        self.files = _RELEASE_CONFIGS['release1']['files']
+
+    def __iter__(self):
+        for filename in self.files:
+            path = os.path.join(self.data_dir, filename)
+            label = [int(c) for c in path.split("_")]
+            yield path, label
+
+
+class LoadYesNoItem(IterDataPipe):
+    def __init__(self, data_pipe):
+        self.data_pipe = data_pipe
+
+    def __iter__(self):
+        for path, label in self.data_pipe:
+            waveform, sample_rate = torchaudio.load(path)
+            yield YesNoItem(path, label, waveform, sample_rate)
+
+
+def get_yesno_dataset(root_dir: str):  # , download=False, url=_RELEASE_CONFIGS["release1"]["url"]):
+    # TODO: download dataset if necessary
+    return LoadYesNoItem(ListYesNoItems(root_dir))
```

Comment:
You can use existing `DataPipe`s
```py
from torch.utils.data import datapipes

dp = datapipes.iter. ListDirFiles(root_dir)
dp = datapipes.iter.Map(dp, fn=loading_fn)
return dp
```
Or, you use the functional API (I prefer this way)
```py
return ListDirFiles(root_dir).map(fn=loading_fn)
```
where `loading_fn` converts file to `YesNoItem`.